lib/utils/parse-json.js

```diff
@@ -0,0 +1,24 @@
+"use strict"
+var parseJSON = module.exports = function (content) {
+  return JSON.parse(stripBOM(content))
+}
+
+parseJSON.noExceptions = function (content) {
+  try {
+    return parseJSON(content)
+  }
+  catch (ex) {
+    return
+  }
+}
+
+// from read-package-json
+function stripBOM(content) {
+  // Remove byte order marker. This catches EF BB BF (the UTF-8 BOM)
+  // because the buffer-to-string conversion in `fs.readFileSync()`
+  // translates it to FEFF, the UTF-16 BOM.
+  if (content.charCodeAt(0) === 0xFEFF) {
```

Comment:
This may have a bit of an awkward effect where states involving non-strong `content` will now throw the "wrong" error ("undefined is not a function"). Are we okay with that? I can see a couple of places in this PR where this seems to be possible.